### PR TITLE
'My location' - Selectable/Updating marker representing users current location

### PR DESCRIPTION
**Pull request by DiscoTim**
This PR adds a selectable and updating location marker representing the users current location.

## Description

The "My Location" marker is a selectable icon from the same list that the search marker uses, including "Hidden". It can be toggled to follow your location in realtime, or it can be dragged to a desired position. It is separate from and has no effect on the scan marker or scanning. It is completely client side and utilizes the existing locationMarker.

This PR should also fix the bug with not being able to reliably retrieve directions. Directions are based on the 'My Location' marker (visible or not) and are queried with definitive start and end locations.

Fixes bug for search follows location url path for sites running in subdirectories.
Fixes bug for searchMarker will be null on /stats and other pages.

Additionally, some of the wording within the location settings was changed for consistency/clarity.

## Motivation and Context

Currently there is no way to track/update your location within the map, and the scan marker is confusing in that many people assume it is related to your location. With spawn point scanning and larger areas being covered this provides a way to determine/set your location in relation to pokemon..etc..

https://github.com/PokemonGoMap/PokemonGo-Map/issues/992

https://github.com/PokemonGoMap/PokemonGo-Map/pull/1033

https://github.com/PokemonGoMap/PokemonGo-Map/pull/619

## How Has This Been Tested?

 This has been tested on Windows and Android, Firefox and Chrome with both geolocation enabled and disable. (Chrome requires SSL for geolocation outside of localhost)

## Screenshots (if appropriate):
nav1
location2
location3

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Shibby7986** [comment]
cool, I'll try this one out.
I might even have a play with adding sharing the location with other users at some point

**Comment by Cubkyle** [comment]
This works perfectly with your #933. If you could add some sort of persistence to it so that I don't have to set the location every time I reopen the page or refresh, it would be perfect.

**Comment by Shibby7986** [comment]
Working fine for me.

**Comment by DiscoTim** [comment]
@Cubkyle just to clarify, are you not using geolocation, "Follow my location" option? you are dragging the marker to set it?

I've updated it so the location is persistent.

**Comment by Cubkyle** [comment]
@DiscoTim Ah, that's correct. I should have been clearer.

**Comment by DiscoTim** [comment]
@Cubkyle okay great, no problem. It should be doing exactly what you want now. thanks for pointing that out.

**Comment by Cubkyle** [comment]
Persistence works, and in my testing I found it was per-user, as expected. Thanks!

**Comment by nunopitbull** [comment]
I have a question:
I'm running the -nsc and -fl because people were changing their location on map and making my bots get banned. Does this still works with those flags?

**Comment by DiscoTim** [comment]
@nunopitbull I've updated with a new commit, and yes it will work with those command switches, and should with any for that matter. 

I think I've fixed the long standing bug with directions as well. I didn't realize that was an existing bug, thanks for pointing that out.

**Comment by Romfrosk** [comment]
i also wonder how i can reactivate that people can have their map follow their location without doing searches.

i'm running -SS and don't want them to try something funny :p 

will test this PR and see how it goes.

**Comment by DiscoTim** [comment]
@Romfrosk2  this or will do that with the  -nsc and -fl  switches, and they would turn on 'follow my location' , its completely client side, it doesn't effect the search. Each client will have their own location marker. Please let us know how the testing goes.

**Comment by Shibby7986** [comment]
sweet, :) I will turn back on the -nsc :P
my web server instance was not doing any searches anyway but still the icon and options are not needed. for me so its cleaner now

File: static/js/map.js [hunk 1]
```diff
@@ -2230,18 +2284,29 @@ $(function () {
...
-            map.panTo(center)
-            searchMarker.setPosition(center)
+        var lng = position.coords.longitude
+        var center = new google.maps.LatLng(lat, lng)
+
+        if (Store.get('geoLocate')) {
+          var baseURL = location.protocol + '//' + location.hostname + (location.port ? ':' + location.port : '')
+          // the search function makes any small movements cause a loop. Need to increase resolution
+          if (getPointDistance(searchMarker.getPosition(), center) > 40) {
+            $.post(baseURL + '/next_loc?lat=' + lat + '&lon=' + lng).done(function () {
```
Comment (jaquer) [inline static/js/map.js]: This is broken for installations running in a subdir. See #619

Comment (DiscoTim) [inline static/js/map.js]: @jaquer really surprised this hasn't been fixed yet. it should be fixed now. caught another bug here where the searchMarker will be null on stats and other pages, but this function will still be running.

**Comment by justsomedudeonthenet** [comment]
This is an excellent feature I never even realized I wanted!

Tested and is working for me.

**Pull request merged**